src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -3512,18 +3513,22 @@ void mvnw(Register Rd, Register Rm,
   void NAME(Condition cond, PRegister Pd, SIMD_RegVariant T, PRegister Pg,             \
             FloatRegister Zn, FloatRegister Zm) {                                      \
     starti;                                                                            \
-    if (fp == 0) {                                                                     \
-      assert(T != Q, "invalid size");                                                  \
-    } else {                                                                           \
-      assert(T != B && T != Q, "invalid size");                                        \
-      assert(cond != HI && cond != HS, "invalid condition for fcm");                   \
+    assert(T != Q, "invalid size");                                                    \
```

Comment:
Please wrap all of this in `#ifdef ASSERT`

Reply:
Thank you for reviewing. Could you please clarify by what exactly you mean by "Please wrap all of this in #ifdef ASSERT"?  Do you mean squashing the if conditions with the asserts? The assert macro calls are already inside a "#define".

Reply:
That whole block is nothing but assertions. It'd be easier to read (and a little quicker to compile) if the block were surrounded by #ifdef ASSERT.

Reply:
I am sorry it's still a bit unclear to me. So we can't add the `#ifdef ASSERT` within a `#define` block. Did you mean something like this -  
```
#ifdef ASSERT
  #define INSN(…)
  assert()
  assert()
  ...
else
  #define INSN(…)
  // no asserts here
  ...
  ...
#undef INSN
#endif
```

Reply:
Ah, sorry. I made a mistake.